src/member.rs

```diff
@@ -725,7 +725,7 @@ where
         loop {
             futures::select! {
                 notification = rx_consensus.next() => match notification {
-                        Some(notification) => self.on_consensus_notification(notification),
+                        Some(notification) => self.on_consensus_notification(notification).await,
```

Comment:
I am not sure if we want to `await` here, we could alternatively add a vector of futures outputting signed data and poll that vector.